frameworks/helloworld/integration/tests/test_sanity.py

```diff
@@ -71,6 +72,37 @@ def test_bump_hello_nodes():
     tasks_not_updated('hello', hello_ids)
 
 
+@pytest.mark.sanity
+def test_lock():
+    '''This test verifies that a second scheduler fails to startup when
+    an existing scheduler is running.  Without locking, the scheduler
+    would fail during registration, but after writing its config to ZK.
+    So in order to verify that the scheduler fails immediately, we ensure
+    that the ZK config state is unmodified.'''
+
+    zk_path = "dcos-service-{}/ConfigTarget".format(PACKAGE_NAME)
+
+    # Get ZK state from running framework
+    zk_config_old = shakedown.get_zk_node_data(zk_path)
+
+    # Install second framework
+    options = {"service": {"name": "hello-world-lock"}, "hello": {"count": 2}}
```

Comment:
Looks like the ZK path is derived from `name` in the service spec yaml. Hello-world's service name is currently hardcoded:

```
dcos-commons/frameworks/helloworld/src/main/dist$ cat svc.yml 
name: "hello-world"
[...]
```

So it should be fine. This might need revisiting if that field stops being hardcoded though. Would it be possible to increase the scheduler task count in marathon, to ensure we get two identical schedulers even if that `name` field becomes templated out (like it probably should be)?